satpy/readers/netcdf_utils.py

```diff
@@ -85,10 +85,12 @@ class NetCDF4FileHandler(BaseFileHandler):
         xarray_kwargs (dict): Addition arguments to `xarray.open_dataset`
         cache_var_size (int): Cache variables smaller than this size.
         cache_handle (bool): Keep files open for lifetime of filehandler.
+            Uses xarray.backends.CachingFileManager, which uses a least
+            recently used cache.
```

Comment:
I've lost track of the caching functionality of this helper class. What are the main differences now between the xarray implementation and the "cached" version specific to Satpy? Is the main difference the support for NetCDF groups and loading per-variable?

Reply:
`CachingFileManager` from xarray uses (by default) xarrays global LRU file cache, which by default keeps at most 128 files open (configurable).  Satpy kept everything open indefinitely. This could have an impact on performance, but is probably a good thing overall. 

By using `CachingFileManager`, the resulting `xarray.DataArray` is serialisable and therefore dask-distributed friendly.

Not sure about your point on groups — groups were already supported?

Reply:
I didn't think xarray's default `xr.open_dataset` supported groups so we had this helper class and added on caching functionality later. I think the idea I was going for when I made the above comment originally was: why can't we just use `xr.open_dataset` instead of this class using `CachingFileManager`? My assumption was that we're supporting NetCDF groups here while xarray doesn't.

Reply:
Oh, do you mean the entire `NetCDF4FileHandler` class is redundant if we use `CachingFileManager` and `xr.open_dataset`s group ability?  It would seem to be a bit more, such as providing a convenient indexing mechanism.  But maybe I still misunderstand.

Reply:
I guess that's what I'm asking, not stating. You're right there is the `__getitem__` logic and the group handling is important enough. I guess I was mostly just curious what the difference was overall.